File: strategy_java/src/test/java/com/kmmania/strategy_training/DefaultTrainingTest.java
package com.kmmania.strategy_training;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class DefaultTrainingTest extends BaseTrainingTest {

    @Test
    @DisplayName("train() method should print DefaultTraining message")
    void train_ShouldPrintDefaultTrainingMessage() {
        // Given
        DefaultTraining defaultTraining = new DefaultTraining();

        // When
        defaultTraining.train();

        // Then
        assertThat(getStandardOutContent()).contains("*** Default training ***");
        assertThat(getStandardOutContent()).contains("...");
    }
}

File: strategy_kotlin/src/test/kotlin/strategy_training/DefaultTrainingTest.kt
package strategy_training

import com.kmmania.strategy_training.DefaultTraining
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DefaultTrainingTest : BaseTrainingTest() {
    @Test
    fun `train() method should print DefaultTraining message`() {
        // Given
        val defaultTraining = DefaultTraining()

        // When
        defaultTraining.train()

        // Then
        assertThat(getStandardOutContent()).contains("*** Default training ***")
        assertThat(getStandardOutContent()).contains("...")
    }
}